app/models/period.rb

```diff
@@ -50,6 +50,7 @@ def initialize(attributes = {})
     else
       value.to_date.beginning_of_month
     end
+    validate!
```

Comment:
Fail fast if a Period is created w/o type or value (which I saw during some of this refactoring).